src/libs/actions/App.js

```diff
@@ -177,33 +167,39 @@ function fixAccountAndReloadData() {
  * pass it in as a parameter. withOnyx guarantees that the value has been read
  * from Onyx because it will not render the AuthScreens until that point.
  * @param {Object} session
+ * @param {string} currentPath
  */
-function setUpPoliciesAndNavigate(session) {
-    Linking.getInitialURL()
-        .then((url) => {
-            if (!url) {
-                return;
-            }
-            const path = new URL(url).pathname;
-            const params = new URLSearchParams(url);
-            const exitTo = params.get('exitTo');
-            const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(url, session.email);
-            const shouldCreateFreePolicy = !isLoggingInAsNewUser
-                        && Str.startsWith(path, Str.normalizeUrl(ROUTES.TRANSITION_FROM_OLD_DOT))
+function setUpPoliciesAndNavigate(session, currentPath) {
+    if (!session || !currentPath || !currentPath.includes('exitTo')) {
+        return;
+    }
+
+    let exitTo;
+    try {
+        const params = new URLSearchParams(currentPath);
+        exitTo = params.get('exitTo');
+    } catch (error) {
+        // URLSearchParams is unsupported on iOS so we catch th error and
+        // silence it here since this is primarily a Web flow
+        return;
+    }
```

Comment:
How did you encounter this error. Why didn't we have an error before?

Reply:
On a side note, the currentPath is not a URL and I wonder if that makes a difference. I did some tests on this string in the JS console and it works fine to access the params except for accessing the accountID, which I don't think we do here.
E.x
```
Neil currentPath =  /transition?accountID=5&email=neil%2Btrand2%40gmail.com&encryptedAuthToken=9BKwuqHcEe%2FMs1kslRzP5bp3duTochIFdP2uuv0Gbzmg6%2BJkbqb4I%2BWFaBdT%2Bxlj9xx6rQ%2FJqynKnl8FwtsILbGa4QilsHS10tVihZiyMpRQCAYyN9u3oZF3izhC63%2Foy4HDfqfeidzoVvNIGtLuM1fSGqXQTMya0Mxppuk6IwnQG5Mh4E0e49fZPW5RMIovKfHOyMh75dN4ike11aN2emz8cXBysmg9VjJN2rNcXlN04DDS5UZhwFzJrNbrCJGKKTi5fsk70X53ogw2hBXiee3N%2FfH6pGRVlN6MBqzjknb3UcUz5K6L6al5EJSFlzvYiXJhOb2lXiiOSOcIPwU01vZ3B0RgI8WkDYhpYW6gOm06%2Fo%2BucO9IHqjJlHfRoVoNDx%2FmSAOIpAM156sdMISDpGUAQ0H1tshQo%2BC9CcJ9SUY2QV%2BiZo5PrrYwTC2Z3OL8c%2FwPA6dFCWnJ8H1gV1CyiOKVrGzlQnzLB6EKNfZJMGzTgWtkdZfBra86IPUsuxkJ0JC7jvS306w3PjhkxQZNZg9dMEWgbCOXKfuxROFzcRdMphVoeLQrhTR8STRVQXOo6eNstl3P4BmLe9X5jITxWk2kzFrePgo%2BPrPmG4DaQVfzuUhi5xBMtsI64QAbWiFllx%2BH3fx9GAC3a3JOa7%2B9lVOs3o5oSJRX1%2F9FYfjPFYtGVzDGpSP9hwrUjSCCZHP%2FZJF9eiY9W6poJtTkke%2BCWAHEhDKANoWwGyJNvCxBVvTrzcmodyt0QrKBXkXRHFvODBgdHkEC99HhxPIiUjoGOymQys%2BLWHUdNc0ImybLBNmz5rNS5TRLa9Tj%2FkdF5tQ3Z%2BVKJBjsjfV8iiS%2B9bWUikm5oJ5t%2FFs5bB1bANbIq71ODsxFFzDiPDyCUkY8w5r%2B&exitTo=workspace%2Fnew&shortLivedToken=CF2A83E7D96D2ABF7C7A5E7E32CB9E6099BC2B4FF801E1F5A9C682050037D7B0E38750E8FC325B270EFF41741709391C0AE662A29B753DE1B96AC3833719C119AC75954486297163B3EC7C4FB00C3CDA4265A2634F330248F6A8EC983ED9A5921FF2D137653C812AD78DEBA32855DA84C1CC126ECCFBBFDA3F40D2AF0D61A1BD7407EFCA472E47E1E7EA153C7530FDDB7DBD30C311E13422F82208AB2EEB52F2D78A7CC9A53A83EC177E50276F3B1376B7482EF1F97577B9999EF6CED61F26F454B17A484661CF0411E33D1DB968D9E09D187AC74BE080CC0F066260259A36DABAB4DE4C8830104CF19AC172087EBB0992D5EC0262C26770A02B7212033C7287B810D51CDE518FF933D6B5758B3FF9043463B65D60450AC7EE60CF97C8B832CA15245A0FEB64422B9FDAD1929F676B1A
```

Reply:
+1

Reply:
Since we were null checking the url and the url for `Linking.getInitialUrl()` is always null on native, iOS never hit this code before.

I'm seeing your `accountID` issue. Looks like we must create a full `URL` object and pull the search query using the `search` property. I'll combine the `currentPath` with a dummy domain to do this.

Reply:
> Since we were null checking the url and the url for `Linking.getInitialUrl()` is always null on native, iOS never hit this code before.

Aaaah ok, that makes sense.
> I'm seeing your `accountID` issue. Looks like we must create a full `URL` object and pull the search query using the `search` property.

At this point accountID doesn't matter but it would be good to have a solution that works for all params in case we need it in the future. Why do we need to create a URL object? We could create a url string and pass in into `URLSearchParams'. I don't know what the convention is here.

> I'll combine the `currentPath` with a dummy domain to do this.

You should use [CONST.ACTIVE_EXPENSIFY_URL](https://github.com/Expensify/App/blob/c5fb8777a18ddc5b3e367d21fc012a474d3760c3/src/CONST.js#L6).

Reply:
I played around with it more in the JS console and it looks like [this SO answer](https://stackoverflow.com/a/66682499/17686499) is the way to go. I don't think the "bugs" they mention are a problem.

Reply:
Nice find I'll try that